code/controllers/subsystem/vis_overlays.dm

```diff
@@ -67,10 +67,14 @@ SUBSYSTEM_DEF(vis_overlays)
 		UnregisterSignal(thing, COMSIG_ATOM_DIR_CHANGE)
 
 /datum/controller/subsystem/vis_overlays/proc/rotate_vis_overlay(atom/thing, old_dir, new_dir)
+	if(old_dir == new_dir)
+		return
 	var/rotation = dir2angle(old_dir) - dir2angle(new_dir)
 	var/list/overlays_to_remove = list()
-	for(var/i in thing.managed_vis_overlays)
+	//the next loop is going to mess with thing.managed_vis_overlays, so iterate on a copy of it
```

Comment:
This comment is inaccurate, this local variable isn't actually a copy

Reply:
You're right, it's superfluous, the `for in` loop already makes and interates on a copy of `things.managed_vis_overlays`
Thanks.